### PR TITLE
[fix][doc] Add more information for producer_request_hold policy

### PR DESCRIPTION
**Pull request by lin-zhao**
Clarifies that producer_request_hold holds the message for at most client operation timeout.

### Motivation

Clarifies the behavior of producer_request_hold policy that raised some confusion in the community.

### Modifications

- Namespaces admin doc includes mention of client operation timeout when explaining producer_request_hold
- Cookbook for retention expiry includes mention of client operation timeout in the example for producer_request_hold

### Documentation

  
- [x ] `doc` 
(Your PR contains doc changes)

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@Technoboy- 
Could you please review this PR from the technical perspective? 
If it is technically correct, then technical writers can review it from the technical writing perspective.
Thank you! 😊

**Comment by MarvinCai** [comment]
According to the `BacklogQuotaManager`'s logic here: https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BacklogQuotaManager.java#L104-L107
I think the behavior should be broker disconnect producer when it detect backlog quota exceeded, and producer will get `ProducerBlockedQuotaExceededError` which is retryable, so it'll keep reconnecting till backlog is cleared: https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java#L1759-L1762
So no data will be send to broker in such base and producer is hold the data till backlog is cleared.
Will let Jiwei confirm.

**Comment by Technoboy-** [comment]
> https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java#L1759-L1762

yes, right.

**Comment by Technoboy-** [comment]
> @Technoboy- Could you please review this PR from the technical perspective? If it is technically correct, then technical writers can review it from the technical writing perspective. Thank you! 😊

LGTM

**Review by Technoboy- — APPROVED** [review]

File: site2/docs/admin-api-namespaces.md [hunk 1]
```diff
@@ -288,7 +288,7 @@ admin.namespaces().getNamespaceReplicationClusters(namespace)
 
 Backlog quota helps the broker to restrict bandwidth/storage of a namespace once it reaches a certain threshold limit. Admin can set the limit and take corresponding action after the limit is reached.
 
-  1.  producer_request_hold: broker holds but not persists produce request payload
+  1.  producer_request_hold: broker holds but not persists produce request payload until client configuration `operationTimeoutMs` is exceeded.
```
Comment (merlimat) [inline site2/docs/admin-api-namespaces.md:291]: This is not exactly correct. First, the broker does not hold the payload: it's the client that will keep retrying up to the `sendTimeout` configured in the producer.

Comment (lin-zhao) [reply site2/docs/admin-api-namespaces.md:291]: @merlimat  you are exactly right that broker actually does **not** hold the message. I will make an update.

**Comment by lin-zhao** [comment]
> According to the `BacklogQuotaManager`'s logic here: https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BacklogQuotaManager.java#L104-L107 I think the behavior should be broker disconnect producer when it detect backlog quota exceeded, and producer will get `ProducerBlockedQuotaExceededError` which is retryable, so it'll keep reconnecting till backlog is cleared: https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java#L1759-L1762 So no data will be send to broker in such base and producer is hold the data till backlog is cleared. Will let Jiwei confirm.

You are exactly right. This is the right logic for this policy. Though I think the documentation for the policy should be really concise without all the details behind the scene.

**Comment by lin-zhao** [comment]
Please review again.

**Review by MarvinCai — APPROVED** [review]

File: site2/docs/admin-api-namespaces.md [hunk 2]
```diff
@@ -288,9 +288,9 @@ admin.namespaces().getNamespaceReplicationClusters(namespace)
 
 Backlog quota helps the broker to restrict bandwidth/storage of a namespace once it reaches a certain threshold limit. Admin can set the limit and take corresponding action after the limit is reached.
 
-  1.  producer_request_hold: broker holds but not persists produce request payload
+  1.  producer_request_hold: the producer holds the message and retries until client configuration `operationTimeoutMs` is exceeded
```
Comment (codelipenghui) [inline site2/docs/admin-api-namespaces.md:291]: It should be until client configuration `sendTimeoutMs` is exceeded.
The `operationTimeoutMs` is for producer creation or consumer creation.

Comment (lin-zhao) [reply site2/docs/admin-api-namespaces.md]: Addressed. Good catch. `operationTimeoutMs` applies for producer creation but `sendTimeoutMs` applies for send requests.

Comment (lin-zhao) [reply site2/docs/admin-api-namespaces.md]: Addressed.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**